### PR TITLE
Web UI: Improve "..." menu

### LINKED ISSUE #9617: Web UI: Add folder "..." menu > Download as zip
**Is your feature request related to a problem? Please describe.**
It took me a while to find out that I can download a folder as zip when selecting it. 

**Describe the solution you'd like**
It would be more intuitive to offer this option via "..." menu similar as the one for single files.

**Describe alternatives you've considered**
Selecting it previously is not as intuitive as visually suggesting actions with the "..." menu.

**Additional context**
No-folder-menu

### PR DESCRIPTION
**Pull request by darkdragon-001**
## Description
- Adds "..." menu for prefixes (folders) to allow downloading as zip (fixes #9617)
- Add hover descriptions (`title` attribute).

## Motivation and Context
More intuitive user interface.

## How to test this PR?
Open Web UI and test.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: browser/app/js/objects/PrefixActions.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/*
+ * MinIO Cloud Storage (C) 2018 MinIO, Inc.
```
Comment (harshavardhana) [inline browser/app/js/objects/PrefixActions.js]:
```suggestion
 * MinIO Cloud Storage (C) 2020 MinIO, Inc.
```

this is a new file

**Review by harshavardhana — CHANGES_REQUESTED** [review]
rest of the changes look good.

**Review by harshavardhana — APPROVED** [review]

**Comment by darkdragon-001** [comment]
In order to complete #5956, I added
- "..." menu: download file
  _For some reason, the fourth items for pictures is shown in the next row:
  next_row
  but I have no idea where the CSS `min-width` comes from..._
- "..." menu: delete folder/prefix

I need clarification for the outstanding points
- "..." menu: share folder/prefix
  -> Is there an RPC method to get something like [Javascript SDK presignedUrl()](https://docs.minio.io/docs/javascript-client-api-reference.html#presignedUrl)
- "+" menu: create empty folder/prefix
  -> Are empty folders supported in the latest version also for _distributed backend_ (#2423 only adds support for _fs_ backend)?

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :x: [more...](#5715941a-220c-438e-b3bb-08b590b19e91) |
<a name="5715941a-220c-438e-b3bb-08b590b19e91"></a>
### 9631-a302531/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:31607
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 498816f667b2:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 1 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 1 minutes and 7 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 25 seconds
(8/15) Running minio-dotnet tests ... done in 28 seconds
(9/15) Running minio-go tests ... done in 31 seconds
(10/15) Running minio-java tests ... done in 28 seconds
(11/15) Running minio-js tests ... FAILED in 22 seconds
{
  "name": "minio-js",
  "function": "getBucketPolicy(bucketName, cb)",
  "args": "bucketName:minio-js-test-9a64b8a9-546b-4308-8c50-1dc9604f89e0",
  "duration": 11,
  "status": "FAIL",
  "error": "S3Error: The bucket policy does not exist at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}
(11/15) Running minio-py tests ... done in 55 seconds
(12/15) Running s3cmd tests ... done in 15 seconds
(13/15) Running s3select tests ... done in 2 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#bcbef46f-ea07-4f22-bf91-efe5f4abc6bb) |
| mint-gateway-nas.sh | :x: [more...](#409da79b-9973-4d2c-ac9b-40c1c2068f9b) |
<a name="bcbef46f-ea07-4f22-bf91-efe5f4abc6bb"></a>
### 9631-d7c8037/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:32122
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 8f0201800c0f:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 7 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 2 minutes and 36 seconds
(4/15) Running aws-sdk-ruby tests ... done in 15 seconds
(5/15) Running awscli tests ... FAILED in 47 seconds
{
  "name": "awscli",
  "duration": 1592,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (BucketAlreadyOwnedByYou) when calling the CreateBucket operation: Your previous request to create the named bucket succeeded and you already own it."
}
(5/15) Running healthcheck tests ... done in 1 seconds
(6/15) Running mc tests ... done in 4 minutes and 30 seconds
(7/15) Running minio-dotnet tests ... done in 1 minutes and 22 seconds
(8/15) Running minio-go tests ... done in 4 minutes and 0 seconds
(9/15) Running minio-java tests ... done in 26 minutes and 55 seconds
(10/15) Running minio-js tests ... done in 2 minutes and 47 seconds
(11/15) Running minio-py tests ... done in 9 minutes and 43 seconds
(12/15) Running s3cmd tests ... done in 2 minutes and 35 seconds
(13/15) Running s3select tests ... done in 58 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="409da79b-9973-4d2c-ac9b-40c1c2068f9b"></a>
### 9631-d7c8037/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:31970
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 09c141736198:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 1 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 2 minutes and 4 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 28 seconds
(8/15) Running minio-dotnet tests ... done in 30 seconds
(9/15) Running minio-go tests ... done in 49 seconds
(10/15) Running minio-java tests ... done in 25 seconds
(11/15) Running minio-js tests ... FAILED in 16 seconds
{
  "name": "minio-js",
  "function": "getBucketPolicy(bucketName, cb)",
  "args": "bucketName:minio-js-test-9a4f1641-1d01-49b5-ae44-24eca1afcb65",
  "duration": 8,
  "status": "FAIL",
  "error": "S3Error: The bucket policy does not exist at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}
(11/15) Running minio-py tests ... done in 56 seconds
(12/15) Running s3cmd tests ... done in 17 seconds
(13/15) Running s3select tests ... done in 2 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by darkdragon-001** [comment]
I also added tests for the newly introduced parts.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :x: [more...](#b15fffe2-3c43-49e7-a982-7748dac7e97d) |
<a name="b15fffe2-3c43-49e7-a982-7748dac7e97d"></a>
### 9631-cfed934/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev3.minio.io:30721
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f5871e0db798:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 1 seconds
(4/15) Running aws-sdk-ruby tests ... done in 1 seconds
(5/15) Running awscli tests ... done in 2 minutes and 5 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 28 seconds
(8/15) Running minio-dotnet tests ... done in 30 seconds
(9/15) Running minio-go tests ... done in 44 seconds
(10/15) Running minio-java tests ... done in 27 seconds
(11/15) Running minio-js tests ... FAILED in 14 seconds
{
  "name": "minio-js",
  "function": "getBucketPolicy(bucketName, cb)",
  "args": "bucketName:minio-js-test-22872ba4-f6f7-422d-94a0-c64c0e6042e9",
  "duration": 11,
  "status": "FAIL",
  "error": "S3Error: The bucket policy does not exist at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}
(11/15) Running minio-py tests ... done in 55 seconds
(12/15) Running s3cmd tests ... done in 15 seconds
(13/15) Running s3select tests ... done in 1 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#1030862a-ce18-44cb-8a90-5c86f479c067) |
<a name="1030862a-ce18-44cb-8a90-5c86f479c067"></a>
### 9631-4549037/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev3.minio.io:31802
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f6d93fc91560:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 7 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 3 minutes and 21 seconds
(4/15) Running aws-sdk-ruby tests ... done in 16 seconds
(5/15) Running awscli tests ... done in 2 minutes and 54 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 2 minutes and 53 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 19 seconds
(9/15) Running minio-go tests ... done in 3 minutes and 43 seconds
(10/15) Running minio-java tests ... done in 26 minutes and 8 seconds
(11/15) Running minio-js tests ... FAILED in 1 minutes and 28 seconds
{
  "name": "minio-js",
  "function": "fPutObject(bucketName, objectName, filePath, metaData, callback)",
  "args": "bucketName:minio-js-test-8a239e98-6e9c-404e-8150-771e87a95f60, objectName:datafile-65-MB, filePath:/tmp/datafile-65-MB, metaData: [object Object]",
  "duration": 953,
  "status": "FAIL",
  "error": "Error: read ECONNRESET at TCP.onStreamRead (internal/stream_base_commons.js:205:27)"
}
(11/15) Running minio-py tests ... done in 9 minutes and 29 seconds
(12/15) Running s3cmd tests ... done in 1 minutes and 22 seconds
(13/15) Running s3select tests ... done in 1 minutes and 0 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by kanagarajkm — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#2ee9cd66-02eb-4fcd-bc96-ea18f95b4a8a) |
| mint-gateway-nas.sh | :x: [more...](#30fe9b48-2fe4-46c6-9ce0-9672a92045a1) |
<a name="2ee9cd66-02eb-4fcd-bc96-ea18f95b4a8a"></a>
### 9631-25bc7de/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:32400
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 6621ea2b0837:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 3 minutes and 0 seconds
(4/15) Running aws-sdk-ruby tests ... done in 16 seconds
(5/15) Running awscli tests ... FAILED in 56 seconds
{
  "name": "awscli",
  "duration": 1579,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (BucketAlreadyOwnedByYou) when calling the CreateBucket operation: Your previous request to create the named bucket succeeded and you already own it."
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 3 minutes and 1 seconds
(7/15) Running minio-dotnet tests ... done in 1 minutes and 23 seconds
(8/15) Running minio-go tests ... done in 3 minutes and 58 seconds
(9/15) Running minio-java tests ... done in 26 minutes and 6 seconds
(10/15) Running minio-js tests ... done in 2 minutes and 36 seconds
(11/15) Running minio-py tests ... done in 9 minutes and 55 seconds
(12/15) Running s3cmd tests ... done in 1 minutes and 21 seconds
(13/15) Running s3select tests ... done in 1 minutes and 1 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="30fe9b48-2fe4-46c6-9ce0-9672a92045a1"></a>
### 9631-25bc7de/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:32409
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp c3100bb8f144:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 1 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 2 minutes and 4 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 25 seconds
(8/15) Running minio-dotnet tests ... done in 30 seconds
(9/15) Running minio-go tests ... done in 51 seconds
(10/15) Running minio-java tests ... done in 32 seconds
(11/15) Running minio-js tests ... FAILED in 24 seconds
{
  "name": "minio-js",
  "function": "getBucketPolicy(bucketName, cb)",
  "args": "bucketName:minio-js-test-cf412802-a459-40bb-8abf-162f293ffe06",
  "duration": 11,
  "status": "FAIL",
  "error": "S3Error: The bucket policy does not exist at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}
(11/15) Running minio-py tests ... done in 1 minutes and 18 seconds
(12/15) Running s3cmd tests ... done in 34 seconds
(13/15) Running s3select tests ... done in 2 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by darkdragon-001** [comment]
Thanks for review and merge!

I would still be glad to see the questions in https://github.com/minio/minio/pull/9631#issuecomment-633025136 answered in order to create a follow-up PR.

**Pull request merged**